Makefile

```diff
@@ -586,32 +643,17 @@ test-shell:
 	./scripts/common/proxy-test.sh
 	./scripts/common/yaml-test.sh
 	./addons/rook/template/test/install.sh
+	./scripts/common/test/common-test.sh
 
-.PHONY: kurl-util-image
-kurl-util-image:
-	docker pull $(KURL_UTIL_IMAGE)
+##@ Release
 
-.PHONY: generate-addons
+.PHONY: generate-addons ## Generate the addons when we deploy to staging and production.
 generate-addons:
 	node bin/generate-addons.js
 
-.PHONY: generate-mocks
-generate-mocks:
-	mockgen -source=pkg/cli/cli.go -destination=pkg/cli/mock/mock_cli.go
-	mockgen -source=pkg/preflight/runner.go -destination=pkg/preflight/mock/mock_runner.go
-
-.PHONY: shunit2
-shunit2: common-test #TODO include other tests
-
-.PHONY: common-test
-common-test:
-	./scripts/common/test/common-test.sh
-
-.PHONY: init-sbom
```

Comment:
Because init-sbom is called in install-spdx-sbom-generator
Then, install-spdx-sbom-generator is called in generate-sbom.

So, would we have any reason to call ` init-sbom` and/or `install-spdx-sbom-generator` directly instead of use `generate-sbom`? if so, then we need to came back the PHONY.

Also, in this case could you please help me out know when both should be used alone?

Reply:
I don't see a use case for this task to be called independently. I think we should get rid of it and add it as part of `install-spdx-sbom-generator` task.

Reply:
unfortunately i dont understand make enough to grok your comment

Reply:
.PHONY is a "don't footgun yourself" thing IIRC?
It should be present for any target that doesn't actually make a file of that name, regardless of how it's called, so that you don't accidentally break `make test` after running `touch test` (etc)

Reply:
Hi @laverya, 

I found its explanation here: https://www.gnu.org/software/make/manual/make.html#Phony-Targets (however, I agree that is not easy to digest). I understand that:

a) We should use PHONY if the name of target can be == a name of file so that we avoid conflicts 
b) If we want to create a target that will call other ones: 

```
all : prog1 prog2 prog3
.PHONY : all

prog1 : prog1.o utils.o
        cc -o prog1 prog1.o utils.o

prog2 : prog2.o
        cc -o prog2 prog2.o

prog3 : prog3.o sort.o utils.o
        cc -o prog3 prog3.o sort.o utils.o
```
c) If you want to ensure that the operation A will only occurs after the operation B finished:

``` 
SUBDIRS = foo bar baz

.PHONY: subdirs $(SUBDIRS)

subdirs: $(SUBDIRS)

$(SUBDIRS):
        $(MAKE) -C $@

foo: baz
```
> Here we’ve also declared that the foo sub-directory cannot be built until after the baz sub-directory is complete; this kind of relationship declaration is particularly important when attempting parallel builds.

So, I think your comment is right and I can go along with:
> It should be present for any target that doesn't actually make a file of that name, regardless of how it's called, so that you don't accidentally break make test after running touch test (etc)

In this case, I reverted the PHONE targets ones. However, I understand that the advantage in this case is: ensure that we run the `init-sbom` first, then run the `install-spdx-sbom-generator` for last run the next steps when we call `generate-sbom`. 

Please, also let me know if you have a diff understand.

Reply:
that's pretty much it!